v2/src/log_sink_etw.c

```diff
@@ -0,0 +1,497 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdbool.h>
+
+#include "windows.h"
+
+#include "TraceLoggingProvider.h"
+#include "evntrace.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/logger.h"
+#include "c_logging/log_sink_if.h"
+#include "c_logging/log_sink_etw.h"
+
+// max number of properties we want to have in the event
+#define LOG_MAX_ETW_PROPERTY_VALUE_PAIR_COUNT 64
+
+// max metadata size
+#define MAX_METADATA_SIZE 4096
+#define MAX_SELF_DESCRIBED_EVENT_WITH_METADATA (sizeof(SELF_DESCRIBED_EVENT) + MAX_METADATA_SIZE)
+
+/* Codes_SRS_LOG_SINK_ETW_01_084: [ log_sink_etw_log shall use as provider GUID DAD29F36-0A48-4DEF-9D50-8EF9036B92B4. ]*/
+TRACELOGGING_DEFINE_PROVIDER(
+    g_my_component_provider,
```

Comment:
>g_my_component_provider [](http://example.com/codeflow?start=4&length=23)

any chance we get a better name than "my"?

Reply:
Renamed.